gsw_xarray/_core.py

```diff
@@ -7,23 +8,77 @@
 from ._names import _names
 from ._check_funcs import _check_funcs
 
+try:
+    import pint_xarray
```

Comment:
Good news is, we have control over our unit strings, so we picked a units style that is compatible with both pint and UDUNITS (there are a few exceptions I don't recall offhand)

Reply:
Oh I see, so far you just set units strings on the output, and ignore units on the input unless they're already pint-arrays. :+1:

Reply:
We aren't quite at the stage of trying to parse inputs, it's planned/desired, and my guess is we would rely heavily on cf-xarray. So correct, right now we just set the correct attributes of the output of the GSW functions.

Basically, the end goal is to have a gsw accessor that automatically lets you know what properties it can calculate for you based on the input dataset. I want it to even do all the right intermediate conversions (e.g. practical sal to SA when asking for density).

Reply:
hah nice! I wrote a neutral density wrapper that only takes a dataset as input and it's great. Caught so many "in_situ_temperature" not "potential_temperature" bugs!